tox.ini

```diff
@@ -0,0 +1,41 @@
+# tox (https://tox.readthedocs.io/) is a tool for running tests
+# in multiple virtualenvs. This configuration file will run the
+# test suite on all supported python versions. To use it, "pip install tox"
+# and then run "tox" from this directory.
+
+[tox]
+envlist = py{35}-django{22}, py{38}-django{22,30}, py38-quality
+
+[testenv:pyNM-djangoNM]
+setenv =
+    SELENIUM_BROWSER=chrome
+
+passenv =
+    *
+
+deps =
+    -rtest-requirements.txt
+    django22: Django==2.2.13
+    django30: Django>=3.0,<3.1
+
+commands_pre =
+    pip install -r {envdir}/src/xblock-sdk/requirements/base.txt
+    pip install -r {envdir}/src/xblock-sdk/requirements/test.txt
+
+    # upgrade both bok-choy and selenium to avoid BrokenPromise Error
+    # caused due to deprecated chromeOptions issue
+    pip install selenium==3.141.0 bok-choy==1.1.1
+
+commands =
+    python run_tests.py
+
+[testenv:py38-quality]
+deps =
+    pep8
+    pylint
+
+commands =
+    pep8 xblock_skytap --max-line-length=120
+    pep8 tests --max-line-length=120
+    pylint xblock_skytap --disable=all --enable=function-redefined,undefined-variable,unused-import,unused-variable
+    pylint tests --disable=all --enable=function-redefined,undefined-variable,unused-import,unused-variable
```

Comment:
Instead of using such a limited set of pylint flags, could you enable them all and fix/disable the individual pylint errors that arise?  I tried that and only found a few, some of which look useful (comments inline):

```bash
py38-quality run-test: commands[2] | pylint xblock_skytap
************* Module xblock_skytap

# Just needs a comment at the top
xblock_skytap/__init__.py:1:0: C0114: Missing module docstring (missing-module-docstring)

************* Module xblock_skytap.exceptions

# remove the pass
xblock_skytap/exceptions.py:10:4: W0107: Unnecessary pass statement (unnecessary-pass)
xblock_skytap/exceptions.py:17:4: W0107: Unnecessary pass statement (unnecessary-pass)

************* Module xblock_skytap.skytap

# add to requirements.txt
xblock_skytap/skytap.py:19:0: E0401: Unable to import 'requests' (import-error)

# quiet with --max-parents=20 or whatever
xblock_skytap/skytap.py:44:0: R0901: Too many ancestors (13/7) (too-many-ancestors)

# Ok to disable these warnings for this line
xblock_skytap/skytap.py:154:21: W0613: Unused argument 'data' (unused-argument)
xblock_skytap/skytap.py:154:27: W0613: Unused argument 'suffix' (unused-argument)

# rearrange import orders
xblock_skytap/skytap.py:20:0: C0411: third party import "from simplejson import JSONDecodeError" should be placed before "import requests" (wrong-import-order)
xblock_skytap/skytap.py:21:0: C0411: third party import "from xblock.core import XBlock" should be placed before "import requests" (wrong-import-order)
xblock_skytap/skytap.py:22:0: C0411: third party import "from xblock.exceptions import JsonHandlerError" should be placed before "import requests" (wrong-import-order)
xblock_skytap/skytap.py:23:0: C0411: third party import "from xblock.fields import Scope, String" should be placed before "import requests" (wrong-import-order)
xblock_skytap/skytap.py:24:0: C0411: third party import "from xblock.fragment import Fragment" should be placed before "import requests" (wrong-import-order)
xblock_skytap/skytap.py:25:0: C0411: third party import "from xblockutils.resources import ResourceLoader" should be placed before "import requests" (wrong-import-order)
xblock_skytap/skytap.py:26:0: C0411: third party import "from xblockutils.settings import XBlockWithSettingsMixin" should be placed before "import requests" (wrong-import-order)
xblock_skytap/skytap.py:27:0: C0411: third party import "from xblockutils.studio_editable import StudioEditableXBlockMixin" should be placed before "import requests" (wrong-import-order)
```

Reply:
To be honest, I saw those quality tests in the CircleCI Build and moved them onto Tox. But you're right, when I ran the pylint with the basic setup, I got so many tweaks that should be done. I'll get that sorted :slightly_smiling_face:

Reply:
Alright, that's done. I didn't add a `requirements.txt`. I just added `requests` to the `tox` dependencies. (for the quality tests, I  wasn't installing the dependencies)